src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ConnectionString.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Rest.ClientRuntime.Azure.TestFramework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// This class represents the connection string being set by the user
+    /// e.g. TEST_CSM_ORGID_AUTHENTICATION="AADTenant=72f98AAD-86f1-2d7cd011db47;ServicePrincipal=72f98AAD-86f1-2d7cd011db47;Password=tzT2+LJBRkSAursui7/Qgo+hyQQ=;SubscriptionId=5562fbd2-HHHH-WWWW-a55d-lkjsldkjf;BaseUri=https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/;GraphUri=https://graph.windows.net/"
+    /// </summary>
+    public class ConnectionString
+    {
+        #region fields
+        private Dictionary<string, string> _keyValuePairs;
```

Comment:
The case manipulation was to honor HttpMockServer dictionary which is currently does not get initialized with IComparer.
But this is a good suggestion of making this dictionary use IComparer.

---
In reply to: [78654692](https://github.com/Azure/azure-sdk-for-net/pull/2328#discussion_r78654692) [](ancestors = 78654692)